### PR TITLE
Base plugin for new layer selector

### PR DESCRIPTION
**Pull request by mmcfarland**
Adds the basic conceptual components to start writing the layer selector
v2 plugin.  These files and contents are not meant to be preserved over
the course of development.

Inculdes:
* Templates
* Local CSS
* Loads JSON layer config

Connects #478

### CODE REVIEW

**Comment by kdeloach** [comment]
+1 Works

File: src/GeositeFramework/plugins/layer_selector_v2/main.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    // https://dojotoolkit.org/documentation/tutorials/1.7/cdn
+    packages: [
+        {
+            name: "jquery",
+            location: "//ajax.googleapis.com/ajax/libs/jquery/1.9.0",
+            main: "jquery.min"
+        },
+        {
+            name: "underscore",
+            location: "//cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3",
```
Comment (caseypt) [inline src/GeositeFramework/plugins/layer_selector_v2/main.js]: Can we start with something a little newer?

Comment (caseypt) [inline src/GeositeFramework/plugins/layer_selector_v2/main.js]: Whoops. This is the newest. My bad.

File: src/GeositeFramework/plugins/layer_selector_v2/main.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    function (declare, PluginBase, layerSourcesJson, templates, $, _) {
+
+        return declare(PluginBase, {
+            toolbarName: "Map Layers v2",
+            fullName: "Configure and control layers to be overlayed on the base map.",
+            toolbarType: "sidebar",
+            allowIdentifyWhenActive: true,
+            hasCustomPrint: true,
+
+
```
Comment (caseypt) [inline src/GeositeFramework/plugins/layer_selector_v2/main.js]: Extra line.

File: src/GeositeFramework/plugins/layer_selector_v2/plugin.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "css": [
+        "plugins/layer_selector_v2/main.css"
+    ],
+    "use": {
+        "underscore": { "attach": "_" }
```
Comment (caseypt) [inline src/GeositeFramework/plugins/layer_selector_v2/plugin.json]: I don't know if it's possible here, but can we avoid use of `use`? In the future, it'll make it easier to transition to the built-in Dojo shimming system.

**Comment by caseypt** [comment]
+1

File: src/GeositeFramework/plugins/layer_selector_v2/plugin.json (hunk 3 above)
Comment (mmcfarland) [inline src/GeositeFramework/plugins/layer_selector_v2/plugin.json]: Underscore is actually AMD compatible now, I'll try and see if I can figure out how to get it without a shim at all.

**Comment by mmcfarland** [comment]
Removed the use of `use` and required the underscore library package directly, cleaned up whitespace.

**Pull request merged**